packages/api/endpoints/granules.js

```diff
@@ -55,6 +61,48 @@ async function list(req, res) {
   return res.send(result);
 }
 
+/**
+ * Create new granule
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} promise of an express response object.
+ */
+const create = async (req, res) => {
+  const {
+    knex = await getKnexClient(),
+    granulePgModel = new GranulePgModel(),
+  } = req.testContext || {};
+
+  let collectionCumulusId;
+  const granule = req.body;
+
+  try {
+    const collectionNameVersion = deconstructCollectionId(granule.collectionId);
+    collectionCumulusId = await getCollectionCumulusId(collectionNameVersion, knex);
+    if (await granulePgModel.exists(knex, {
```

Comment:
This just occurred to me: why is your work checking PosgreSQL instead of DynamoDB to see if the granule already exists. The code on the `master` branch is still RDS phase 1 where DynamoDB is the source of truth. It's not until RDS phase 2, which has our work in a feature branch, where we would need to check PostgreSQL instead of DynamoDB.

Reply:
faae13024c